### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by bjorn3**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,10 +22,9 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 
 open = "1.4.0"
-promising-future = "0.2"
 syntect = "4.2"
 horrorshow = "0.8"
-cgraph = { git = "https://github.com/oli-obk/cgraph.git", rev = "7348380f014fe87338cdcd4d2f22cbda771ae21d" }
+cgraph = { git = "https://github.com/oli-obk/cgraph.git", rev = "5e6a6d4527522609772a9a9a1565b74c9bfe1560" }
```
Comment (DJMcNab) [inline Cargo.toml:27]: I can confirm that https://github.com/oli-obk/cgraph/commit/5e6a6d4527522609772a9a9a1565b74c9bfe1560 is the current master commit.

File: src/main.rs [hunk 2]
```diff
@@ -163,14 +163,14 @@ impl PrirodaSender {
...
         F: FnOnce(&mut PrirodaContext<'_, '_>) -> T + Send + 'static,
     {
-        let (future, promise) = future_promise();
+        let (tx, rx) = mpsc::sync_channel(0);
         let sender = self.0.lock().unwrap_or_else(|err| err.into_inner());
         match sender.send(Box::new(move |pcx: &mut PrirodaContext<'_, '_>| {
-            promise.set(f(pcx));
+            tx.send(f(pcx)).unwrap();
         })) {
-            Ok(()) => match future.value() {
```
Comment (DJMcNab) [inline src/main.rs:171]: Certainly for reviewing purposes, this whole function is a bit messy in terms of control flow. I do believe that these changes are equivalent.

**Review by DJMcNab — APPROVED** [review]
These changes make sense to me

File: src/main.rs (hunk 2 above)
Comment (bjorn3) [reply src/main.rs:171]: Opened #47.

**Pull request merged**